spec/UserDbSpec.js

```diff
@@ -13,45 +13,97 @@ describe("User DB", () => {
-    it("shall be able to retrieve a user with a correct username and password", async function() {
+    it("shall be able to retrieve a user with a correct username and password", async function () {
         let res = await UserDb.addUser("vcedgar", "1234");
-        expect(res).toBe(true);
+        expect(res)
+            .toBe(true);
 
         let result1 = await UserDb.userLogin("vcedgar", "1234");
         let result2 = await UserDb.userLogin("vcedgar", "12345");
         let result3 = await UserDb.userLogin("vcedgar6", "1234");
-        expect(result1).toBe(true);
-        expect(result2).toBe(false);
-        expect(result3).toBe(false);
+        expect(result1)
+            .toBe(true);
+        expect(result2)
+            .toBe(false);
+        expect(result3)
+            .toBe(false);
     });
 
     /* A specification */
-    it("shall be able to add a user with a new username", async function() {
+    it("shall be able to add a user with a new username", async function () {
         let result1 = await UserDb.addUser("zlmonroe", "1234");
         let result1p2 = await UserDb.userLogin("zlmonroe", "1234");
 
-        expect(result1 && result1p2).toBe(true);
+        expect(result1 && result1p2)
+            .toBe(true);
 
         let result3 = await UserDb.addUser("zlmonroe1", "1234");
         let result3p2 = await UserDb.userLogin("zlmonroe1", "1234");
-        expect(result3 && result3p2).toBe(true);
+        expect(result3 && result3p2)
+            .toBe(true);
     });
 
     /* A specification */
-    it("shall fail at adding a user with an existing username", async function() {
+    it("shall fail at adding a user with an existing username", async function () {
         let result1 = await UserDb.addUser("jebush", "1234");
         let result1p2 = await UserDb.userLogin("jebush", "1234");
 
-        expect(result1 && result1p2).toBe(true);
+        expect(result1 && result1p2)
+            .toBe(true);
 
         let result2 = await UserDb.addUser("jebush", "12345");
         let result2p2 = await UserDb.userLogin("jebush", "12345");
-        expect(result2).toBe(false);
-        expect(result2p2).toBe(false);
+        expect(result2)
+            .toBe(false);
+        expect(result2p2)
+            .toBe(false);
 
         let result3 = await UserDb.addUser("jebush1", "1234");
         let result3p2 = await UserDb.userLogin("jebush1", "1234");
-        expect(result3 && result3p2).toBe(true);
+        expect(result3 && result3p2)
+            .toBe(true);
+    });
+
+    /* A specification */
+    it("shall return count of all students in the db", async function () {
+        await UserDb.addUser("test1", "test1");
+        await UserDb.addUser("test2", "test2", "student");
+        await UserDb.addUser("test3", "test3", "teacher");
+
+        const result1 = await UserDb.getUserCount();
+        const result2 = await UserDb.getUserCount("student");
+
+
+        expect(result1)
+            .toBe(result2);
```

Comment:
This expect is bad. You are comparing the usercount function to itself. If there was some error that overlapped both calls (i.e. if both returned 0 because they didn't work) it would pass

Reply:
So I originally tried to do it this way, but for some reason, every time the spec ran, it produced different results because of the tests adding users. I added removeAll before the test started and that still didn’t work

Reply:
The way these tests are set up in general is bad. Every test should be cleaned up after it runs. Unit test functions should be independent. It wouldn't be that hard to do - jasmine has a afterEach type function. (I don't recall the exact name at the moment though)

Reply:
Tried that, for some reason it still produced a different result.

Reply:
This was not resolved. You are still comparing it to itself